### DISCUSSION TITLE
Storybook errors during migration to latest version.

### DISCUSSION THREAD

**Discussion by di-void** [storybookjs/storybook#24829] [Help] [answered]
Hello friends. In our monorepo codebase, we've been using storybook version `7.3.2` for a while now. Everything worked as expected. But the warnings to upgrade to the latest version started to disturb me a bit so I decided to migrate to the latest version by using the command provided: `npx storybook@latest upgrade`.

Everything installed properly without errors and was upgraded but when I ran the `storybook` command to open the manager, it showed me a bunch of errors like this:
```
Could not resolve "@storybook/core-events/preview-errors"

    ../../node_modules/@storybook/preview-api/dist/chunk-ZVJD52XQ.mjs:6:42:
      6 │ import { MissingStoryAfterHmrError } from '@storybook/core-events/preview-errors';
        ╵                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  The path "./preview-errors" is not exported by package "@storybook/core-events":

    ../../node_modules/@storybook/preview-api/node_modules/@storybook/core-events/package.json:23:13:
      23 │   "exports": {
         ╵              ^

  You can mark the path "@storybook/core-events/preview-errors" as external to exclude it from the
  bundle, which will remove this error.
  ```
  Since the project is a vite project, it automatically set the builder to vite.
  I've tried all sorts of things like modifying the project's vite config file according to GPT to exclude it from the bundle like so:
  ```js
  import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  optimizeDeps: {
    exclude: ['@storybook/core-events/preview-errors'],
  },
})
  ```
  That didn't work.
  
  The only thing that worked was following the stack trace to this file in the monorepo's node modules folder: ` ../../node_modules/@storybook/preview-api/dist/chunk-ZVJD52XQ.mjs` and commenting out this import:
  ```js
//  import { MissingStoryAfterHmrError } from '@storybook/core-events/preview-errors';
```
  After commenting it out, I tried to start the manager and it worked. The problem is I'm not the only one collaborating on the project and I don't expect all my teammates to all dive into the monorepo's node modules to do some silly hack to fix it. Plus, I presume that ideally, this shouldn't be how it's supposed to work.
  
  Please any help with this will be much appreciated 🙏🏾

**Comment by Marklb** [comment]
Have you tried deleting your `node_modules` and `package-lock.json` (*or whatever package manager lock file you have*) then reinstalling deps? I typically do that anytime I have an issue after upgrading a dependency. The only way I know to cause that is to install an old `@storybook/core-events` and a newer `@storybook/preview-api`.

If so, could you check that your `@storybook/*` package versions in `node_modules` to be sure they match or check your lock file for any potential conflicts causing multiple `@storybook/core-events` versions to get installed? I haven't checked what specific version added the `./preview-errors` export, but I think it was around v7.4 that it was added.

**Reply by yannbf** [reply, marked as answer]
Thank you so much Marklb, this should fix their use case. This issue can happen when certain Storybook packages are not upgraded, given that they don't follow the same versioning strategy as others, for instance:
```json
{
  "@storybook/react": "7.3.1",
  "@storybook/addon-coverage": "0.0.9"
}
```

and then `storybook upgrade` will update `@storybook/react` and its dependencies, however the lock file won't change the older versions of `@storybook/addon-coverage` dependencies, causing Storybook to end up having multiple versions for the same packages.

@Di-void please report back and mark this as answer if this fixes your issue!

**Reply by di-void** [reply]
Roger. I'll do that, thank you very much.

**Reply by joevaugh4n** [reply]
Glad you solved this, everyone!

**Comment by di-void** [comment]
Thanks so much for your reply. I will try deleting the `node_modules` and `yarn.lock` file (we use yarn) to see if it works.

**Comment by di-void** [comment]
Thank you very much @Marklb and @yannbf. Deleting my `node_modules` and `yarn.lock` file and reinstalling solve the issue for me.
